custom/openclinica/management/commands/odm_to_app.py

```diff
@@ -0,0 +1,198 @@
+            yield StudyEvent(se_def, self.meta)
+
+    @staticmethod
+    def new_subject_module():
+        module = Module.new_module('Study Subjects', None)
+        module.case_type = 'subject'
+        reg_form = module.new_form('Register Subject')
+        reg_form.actions.open_case = OpenCaseAction(name_path="/data/name", external_id=None)
+        reg_form.actions.open_case.condition.type = 'always'
+        return module
+
+    def get_app(self, domain_name, app_name, version=APP_V2):
+        if self._app is None:
+            self._app = Application.new_app(domain_name, app_name, application_version=version)
+            self._app.name = self.name
+            self._app.add_module(self.new_subject_module())
+            for event in self.iter_events():
+                module = event.get_module()
+                for study_form in event.iter_forms():
+                    study_form.add_form_to_module(module)
+                self._app.add_module(module)
+        return self._app
+
+
+class StudyEvent(StudyObject):
+
+    def __init__(self, defn, meta):
+        super(StudyEvent, self).__init__(defn, meta)
+        self.oid = defn['OID']
+        self.name = defn['Name']
+        self.is_repeating = defn['Repeating'] == 'Yes'
+        self.event_type = defn['Type']  # Scheduled, Unscheduled, or Common
+        self._module = None
+
+    def iter_forms(self):
+        for form_ref in self.defn.xpath('./odm:FormRef', namespaces=odm_nsmap):
+            form_oid = form_ref['FormOID']
+            form_def = self.meta.xpath('./odm:FormDef[@OID="{}"]'.format(form_oid), namespaces=odm_nsmap)[0]
+            yield StudyForm(form_def, self.meta)
+
+    def get_module(self):
+        """
+        Return a CommCare module
+        """
+        if self._module is None:
+            self._module = Module.new_module(self.name, None)
+            self._module.case_type = 'event'
+        return self._module
+
+
+class StudyForm(StudyObject):
+
+    def __init__(self, defn, meta):
+        super(StudyForm, self).__init__(defn, meta)
+        self.oid = defn['OID']
+        self.name = defn['Name']
+        self.is_repeating = defn['Repeating'] == 'Yes'
+        self._form = None
+
+    def iter_item_groups(self):
+        for ig_ref in self.defn.xpath('./odm:ItemGroupRef', namespaces=odm_nsmap):
+            ig_oid = ig_ref['ItemGroupOID']
+            ig_def = self.meta.xpath('./odm:ItemGroupDef[@OID="{}"]'.format(ig_oid), namespaces=odm_nsmap)[0]
+            yield ItemGroup(ig_def, self.meta)
+
+    def add_form_to_module(self, module):
+        """
+        Add a CommCare form based in this form's item groups and items to a given CommCare module
+        """
+        if self._form is None:
+            self._form = module.new_form(self.name, None)
+            self._form.source = self.build_xform()
+        else:
+            module.add_insert_form(None, self._form)
```

Comment:
`add_form_to_module` only seems to be called once per instance so I don't think it needs the `if else`